dpl/api/local_announce.py

```diff
@@ -0,0 +1,75 @@
+"""
+This module contains an implementation of everpl announcement logic for
+local networks
+"""
+
+import socket
+
+import zeroconf
+
+
+class LocalAnnounce(object):
+    """
+    This class controls the announcement of everpl instance presence in the
+    local network using Zeroconf (Avahi / Bonjour) protocol
+    """
+    _service_type = "_everpl._tcp.local."
+
+    def __init__(self):
+        """
+        Initializes Zeroconf instance and starts corresponding threads
+        """
+        self._zeroconf = zeroconf.Zeroconf()
+
+    def create_server(
+            self, instance_name: str = None,
+            address: str = "127.0.0.1", port: str = 10800,
+            server_host: str = None
+    ) -> None:
+        """
+        Initializes zeroconf service (everpl hub) announcement with the
+        specified optional params
+
+        :param instance_name: the name of this service (instance, hub);
+               set to the "everpl hub @ hostname._everpl.tcp" by default
+        :param address: an IP address of this instance; set to the 127.0.0.1
+               by default
+        :param port: the port of announced service; set to the 10800 by default
+        :param server_host: hostname of this instance; set to the
+               "hostname.local" by if not set specifically (if server_host is
+               None or is equal to the "0.0.0.0" address)
+        :return: None
+        """
+        hostname = socket.gethostname()
+
+        if instance_name is None:
+            instance_name = "everpl hub @ %s.%s" % (
+                hostname, self._service_type
+            )
+
+        coded_address = socket.inet_aton(address)
+
+        if server_host is None or server_host == "0.0.0.0":
```

Comment:
Codacy false-positive: Possible binding to all addresses

Why it's a _false_-positive: This code does exactly opposing - sets the fully qualified name for a service to be announced to something _different_ from 0.0.0.0. It's not a hardcoded binding to 0.0.0.0 and listening on all interfaces by default.